### PR TITLE
Add in guidelines for contributing

### PR DESCRIPTION
**Pull request by andymckay**
👋 This adds in a couple of things to this repo:

* A code of conduct
* A rather more detailed contributing document for all the awesome people working on this
* Guidelines on the pull request on the contribution document so people will know whats accepted on this repository.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Thank you for sending in this pull request. Please make sure you take a look at the [contributing file](CONTRIBUTING.md). Here's a few things for you to consider out on this pull request:
+
+- [ ] Include a good description of the workflow.
+- [ ] Links to the language or tool will be nice (unless its really obvious)
```
Comment (mscoutermarsh) [inline .github/pull_request_template.md]:
```suggestion
- [ ] Links to the language or tool will be nice (unless its really obvious).
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+## Contributing
+
+[code-of-conduct]: CODE_OF_CONDUCT.md
+
+Hi there 👋 We are excited that you want to contribute a new workflow to this repo. By doing this you are helping our users get up and running with GitHub Actions and that's cool 😎
```
Comment (mscoutermarsh) [inline CONTRIBUTING.md]:
```suggestion
Hi there 👋 We are excited that you want to contribute a new workflow to this repo. By doing this you are helping people get up and running with GitHub Actions and that's cool 😎.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+Contributions to this project are [released](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) to the public under the [project's open source license](LICENSE.md).
+
+Please note that this project is released with a [Contributor Code of Conduct][code-of-conduct]. By participating in this project you agree to abide by its terms.
+
+There are few requirements for adding in a new workflow, which I hope make sense and we'll need to review them before we merge a workflow. There are some more on the pull request template, when you create that, but generally an template:
```
Comment (mscoutermarsh) [inline CONTRIBUTING.md]:
```suggestion
There are few requirements for adding in a new workflow, which we'll need to review before we merge:
```

**Comment by okuryu** [comment]
Is this only for the starter-workflows? I think creating a `.github` repo is better if it go for all repos of actions org.
https://help.github.com/en/articles/creating-a-default-community-health-file-for-your-organization

**Comment by andymckay** [comment]
Good question @okuryu 🙇  The contributing guidelines for starter workflows is different enough that I think we should just go with this one. It's a good idea to add one for the org though so all the other repos get it though.

**Comment by okuryu** [comment]
It may be good to put `CONTRIBUTING.md` for each repository. I think that one `CODE_OF_CONDUCT.md` is enough for the organization because it's rare to need multiple the code of conduct in one organization.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Thank you for sending in this pull request. Please make sure you take a look at the [contributing file](CONTRIBUTING.md). Here's a few things for you to consider out on this pull request:
```
Comment (ethomson) [inline .github/pull_request_template.md]: "Here's a few things for you to consider in this pull request:" ?

**Comment by ethomson** [comment]
I've been wondering if our sample workflows should have some comments in them?  Some are super straightforward and won't need them but some might.  For example:

```yaml
    - name: Setup .NET Core
      uses: actions/setup-dotnet@v1
      with:
        dotnet-version: 2.2.108
```

Could use some information about what's permissible in `dotnet-version`.

Is it worth calling this out in the suggestions?  eg in the pull request template:

```markdown
- [ ] Include comments for any parts that are not obvious or could use clarification
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+
+- [ ] Include a good description of the workflow.
+- [ ] Links to the language or tool will be nice (unless its really obvious).
+- [ ] For the `*.yml` file, please include a matching `ci/properties/*.properties.json` file.
+- [ ] Does not use an Action that isn't in the `actions` organization.
+- [ ] Does not send data to any 3rd party service except for the purposes of installing dependencies.
+- [ ] Does use a paid service or product.
```
Comment (ethomson) [inline .github/pull_request_template.md]: - [ ] Use title case for the names of workflows and steps, for example "Run tests".

Comment (ethomson) [reply .github/pull_request_template.md]:
```yaml
- [ ] The name of CI workflows should only be the name of the language or platform: for example "Go" (not "Go CI" or "Go Buil
- [ ] CI workflows should run `on: [push]`.  Packaging workflows should run:

    on:
        release:
            types: [created]
```

Comment (ethomson) [reply .github/pull_request_template.md]: Oh, also, should this be "Does not use a paid service or product"?

File: .github/pull_request_template.md (hunk 4 above)
Comment (andymckay) [reply .github/pull_request_template.md]: Oops, thanks.

**Comment by andymckay** [comment]
Thanks for the feedback @ethomson some good catches there, I've update the PR template a bit and broken it down into three groups: what to write in the PR, how to write the workflow and properties file and some general notes.

**Comment by andymckay** [comment]
> I think that one CODE_OF_CONDUCT.md is enough for the organization because it's rare to need multiple the code of conduct in one organization.

Good call, we should do that.

**Comment by andymckay** [comment]
Fixes #91

**Pull request merged**